var/spack/repos/builtin/packages/libzmq/package.py

```diff
@@ -70,23 +70,23 @@ class Libzmq(AutotoolsPackage):
 
     # Fix aggressive compiler warning false positive
     patch(
-        "https://github.com/zeromq/libzmq/commit/92b2c38a2c51a1942a380c7ee08147f7b1ca6845.patch?full_index=1",
+        "https://github.com/zeromq/libzmq/pull/4213.patch?full_index=1",
```

Comment:
For my curiosity: What is the rationale behind this change?

Reply:
Just a shorter URL. The patch itself is the same and we checksum the download so we don't need to specify the commit hash.

Reply:
I would stick to the commit, idk you if you can push to a branch of a merged PR, but in principle a PR is a moving target

Reply:
An open PR is a moving target (this has bit me in the past) but I don't think a merged PR is. I don't have strong opinions either way. If we decide Spack should have strong opinions we should add an audit check for this like we did for `?full_index=1` and other such things.

Reply:
I personally have no strong preference whether to use `pull/*.patch?full_index=1` or `commit/*.patch?full_index=1` urls. Once the project comes up with a policy here, I am happy to align.

Thx for the clarification. I just wished you put this rationale in the commit msg, or better made a separate commit. (But then this project **squash-merges** PRs anyways ... :crying_cat_face: :stuck_out_tongue_winking_eye:)